### PR TITLE
LXD Profile acceptance tests

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following acceptance test provides a generic way to see
if deploying LXD profiles generate errors. If no errors are found
it will then inspect the applications to locate any potential charm
profiles, then validating that they have been applied.

## QA steps

```
mkdir /tmp/test-run
mkdir /tmp/artifacts
export JUJU_HOME=/tmp/test-run
export JUJU_REPOSITORY=$GOPATH/src/github.com/juju/juju/acceptancetests/repository
vim $JUJU_HOME/environments.yaml
```

Local LXD environments.yaml
```yaml
environments:
    lxd:
        type: lxd
        test-mode: true
        default-series: bionic
```

Run the following (substitute the right arg):
```
cd acceptancetests
./assess_deploy_lxd_profile_bundle.py
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/B3HYzes2/1083-1813044-follow-on-acceptance-tests

File: acceptancetests/repository/bundles-lxd-profile.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+series: bionic
+machines:
+  '0': {}
+  '1': {}
+  '2': {}
+  '3': {}
+applications:
+  lxd-profile:
+    charm: cs:~juju-qa/bionic/lxd-profile-without-devices-0
+    num_units: 4
```
Comment (hmlanigan) [inline acceptancetests/repository/bundles-lxd-profile.yaml]: Can we make the unit count 8 and deploy the other 4 to the machines.  More than 1 unit per machine seems very common in our failure scenarios

Comment (SimonRichardson) [reply acceptancetests/repository/bundles-lxd-profile.yaml]: Yes, good shout

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**